src/PHPUnitCommand.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\NoLeaks\PHPUnit;
+
+use PHPUnit\TextUI\Command;
+use PHPUnit\TextUI\TestRunner;
+use function array_merge;
+
+final class PHPUnitCommand extends Command
+{
+    public static function main(bool $exit = true) : int
+    {
+        $command = new static();
+        return $command->run($_SERVER['argv'], $exit);
+    }
+
+    protected function handleArguments(array $argv) : void
+    {
+        parent::handleArguments($argv);
+        if (! $this->isAutoConfigureEnabled()) {
+            return;
+        }
+        $this->arguments['listeners'] = array_merge(
```

Comment:
Is 'listener' enough? What about extensions?

Reply:
Extension is registered in `createRunner` see line 35. I could not find a proper way to inject params in PHPUnit. Don't forget, it's more a PoC, and needs more investigation. I can do better and write tests if the idea looks good for you.

Reply:
If we can completely drop the `phpunit.xml` customisation and make it work, then that's enough :-)

See `test/e2e`: if those work without the extensions being registered manually, we're golden

Reply:
Great let me a day or two ;) Thank you.